stack-example/README.md

```diff
@@ -0,0 +1,77 @@
+This setup shows how to index haskell packages using `stack`
+and the [haskell-indexer-plugin][haskell-indexer-plugin].
+
+[haskell-indexer-plugin]: https://github.com/google/haskell-indexer
+
+First change directory to `stack-example` and build the plugin.
+```bash
+$ cd stack-example
+$ stack build haskell-indexer-plugin
+```
+
+Then generate a ghc wrapper script. This script will pass
+to GHC all the flags needed to enable the indexer plugin.
+```bash
+$ ./gen-ghc-wrapper.sh
+
+USAGE
+
+    ./gen-ghc-wrapper.sh OUTDIR GHC_PATH PLUGIN_DB
+
+Generates a wrapper for a ghc compiler in OUTDIR that indexes
+using the given compiler.
+
+EXAMPLE
+
+    ./gen-ghc-wrapper.sh ghc_wrapper "$(stack path --compiler-exe)" \
+    "$(stack path --snapshot-pkg-db)"
+```
+
+```bash
+$ ./gen-ghc-wrapper.sh ghc_wrapper "$(stack path --compiler-exe)" \
+  "$(stack path --snapshot-pkg-db)"
+```
+
+Now we can index a package and almost all of its dependencies with
+```bash
+$ export INDEXER_OUTPUT_DIR=/tmp/indexer-output
+$ PATH=$(pwd)/ghc_wrapper:$PATH \
+  STACK_ROOT=$HOME/.stack-indexer stack --system-ghc build <pkg>
+```
+We direct `stack` to use the `ghc` wrapper by incluiding it in the
+`PATH` and passing the flag `--system-ghc` to it.
+
+The purpose of using `STACK_ROOT` above, is to rebuild and index even 
+the dependencies of `haskell-indexer-plugin`. If you are not
+interested in those, then using `STACK_ROOT` can be omitted.
+
+Packages which use Safe Haskell [can't be indexed currently][sh-issue].
+They can be skipped with
+```bash
+$ INDEXER_PLUGIN_ENABLE=0 PATH=$(pwd)/ghc_wrapper:$PATH \
```

Comment:
How does this work?

Reply:
`INDEXER_PLUGIN_ENABLE=0` omits passing `-fplugin Haskell.Indexer.Plugin` to GHC. The wrapper still feeds `-package-db` and `-plugin-package` so GHC is able to find it because it has been used in dependencies.

Reply:
Oh, I should refer to https://ghc.haskell.org/trac/ghc/ticket/15940 here.